satpy/readers/hsaf_h5.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import xarray as xr
+
+from satpy import CHUNK_SIZE
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+
+
+class HSAFFileHandler(BaseFileHandler):
+    """File handler for HSAF H5 files."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init the file handler."""
+        super(HSAFFileHandler, self).__init__(filename,
+                                              filename_info,
+                                              filetype_info)
+
+        self._msg_datasets = {}
+        self._start_time = None
+        self._end_time = None
+        analysis_time = self._get_datetime(filename)
+        self._analysis_time = analysis_time
+
+    @staticmethod
+    def _get_datetime(filename):
+        fname = os.path.basename(filename)
+        dtstr = fname.split('_')[1]
+        return datetime.strptime(dtstr, "%Y%m%d")
+
+    @property
+    def analysis_time(self):
+        """Get validity time of this file."""
+        return self._analysis_time
+
+    def _prepare_variable_for_palette(self, msg, ds_info):
+        colormap = np.array(np.array(msg))
+        return xr.DataArray(colormap, attrs=ds_info, dims=('idx', 'RGB'))
+
+    def get_metadata(self, msg, name):
+        """Get the metadata."""
+        if name == 'SC':
+            ds_info = {
+                'name': 'SC',
+                'filename': self.filename,
+                'data_time': self._analysis_time,
+                'nx': msg.shape[1],
+                'ny': msg.shape[0]
+            }
+        elif name == 'SC_pal':
+            ds_info = {
+                'name': 'SC_pal'
+            }
+        return ds_info
+
+    def get_area_def(self, dsid):
+        """Area definition for h10 SC dataset.
+
+        Since it is not available in the HDF5 message,
+        using hardcoded one (it's known).
+        """
+        if dsid['name'] == 'SC':
+            return self._get_area_def()
+        raise NotImplementedError
+
+    def _get_area_def(self):
+        """Area definition for h10 - hardcoded.
+
+        Area definition not available in the HDF5 message,
+        so using hardcoded one (it's known).
+
+        hsaf_h10:
+          description: H SAF H10 area definition
+          projection:
+            proj: geos
+            lon_0: 0
+            h: 35785831
+            x_0: 0
+            y_0: 0
+            a: 6378169
+            rf: 295.488065897001
+            no_defs: null
+            type: crs
+          shape:
+            height: 916
+            width: 1902
+          area_extent:
+            lower_left_xy: [-1936760.3163240477, 2635854.280233425]
+            upper_right_xy: [3770006.7195370505, 5384223.683413638]
+            units: m
+        """
+        fd_def = get_area_def('msg_seviri_fes_3km')
+        hsaf_def = fd_def[62:62+916, 1211:1211+1902]
+
+        return hsaf_def
+
+    def _get_message(self, idx):
```

Comment:
Better name for this method would be `_get_dataset()` or something. `message` is something bufr/grib files have.